src/Model/Photo.php

```diff
@@ -1017,16 +1038,129 @@ public static function upload(int $uid, array $files)
+		$picture['albumpage']   = DI::baseUrl() . '/photos/' . $user['nickname'] . '/image/' . $resource_id;
+		$picture['picture']     = DI::baseUrl() . '/photo/{$resource_id}-0.' . $Image->getExt();
+		$picture['preview']     = DI::baseUrl() . '/photo/{$resource_id}-{$smallest}.' . $Image->getExt();
+
+		$Image->__destruct();
 
 		Logger::info('upload done', ['picture' => $picture]);
 		return $picture;
 	}
+
+	/**
+	 *
+	 * @param int   $uid   User ID
+	 * @param array $files uploaded file array
+	 * @return string avatar resource
+	 */
+	public static function uploadAvatar(int $uid, array $files)
+	{
+		$data = self::uploadImage($files);
+		if (empty($data)) {
+			return [];
+		}
+
+		$Image    = $data['image'];
+		$filename = $data['filename'];
+		$width    = $data['width'];
+		$height   = $data['height'];
+
+		$resource_id = Photo::newResource();
+		$album       = DI::l10n()->t(Photo::PROFILE_PHOTOS);
+
+		// upload profile image (scales 4, 5, 6)
+		logger::info('starting new profile image upload');
+
+		if ($width > 300 || $height > 300) {
+			$Image->scaleDown(300);
+		}
+
+		$r = Photo::store($Image, $uid, 0, $resource_id, $filename, $album, 4, Photo::USER_AVATAR);
+		if (!$r) {
+			logger::notice('profile image upload with scale 4 (300) failed');
+		}
+
+		if ($width > 80 || $height > 80) {
+			$Image->scaleDown(80);
+		}
+
+		$r = Photo::store($Image, $uid, 0, $resource_id, $filename, $album, 5, Photo::USER_AVATAR);
+		if (!$r) {
+			logger::notice('profile image upload with scale 5 (80) failed');
+		}
+
+		if ($width > 48 || $height > 48) {
+			$Image->scaleDown(48);
+		}
+
+		$r = Photo::store($Image, $uid, 0, $resource_id, $filename, $album, 6, Photo::USER_AVATAR);
+		if (!$r) {
+			logger::notice('profile image upload with scale 6 (48) failed');
+		}
+
+		$Image->__destruct();
+		logger::info('new profile image upload ended');
+
+		$condition = ["`profile` AND `resource-id` != ? AND `uid` = ?", $resource_id, $uid];
+		Photo::update(['profile' => false, 'photo-type' => Photo::DEFAULT], $condition);
+
+		Contact::updateSelfFromUserID($uid, true);
+
+		// Update global directory in background
+		Profile::publishUpdate($uid);
+
+		return $resource_id;
+	}
+
+	/**
+	 *
+	 * @param int   $uid   User ID
+	 * @param array $files uploaded file array
+	 * @return string avatar resource
+	 */
+	public static function uploadBanner(int $uid, array $files)
+	{
+		$data = self::uploadImage($files);
+		if (empty($data)) {
+			Logger::info('upload failed');
+			return [];
+		}
+
+		$Image    = $data['image'];
+		$filename = $data['filename'];
+		$width    = $data['width'];
+		$height   = $data['height'];
+
+		$resource_id = Photo::newResource();
+		$album       = DI::l10n()->t(Photo::BANNER_PHOTOS);
+
+		if ($width > 960) {
+			$Image->scaleDown(960);
+		}
+
+		$r = self::store($Image, $uid, 0, $resource_id, $filename, $album, 3, self::USER_BANNER);
+		if (!$r) {
+			logger::notice('profile banner upload with scale 3 (960) failed');
```

Comment:
Should we really go on with updating the photo entry even the image store failed?
I'd prefer an exception like above

Reply:
The current functionality that uses this function isn't prepared for exceptions and we are deep in the RC phase. So I would appreciate to keep it that way.

Reply:
OK, I see and you're right - we should prepare for a upcoming release
But I'd like to have a TODO at both functions with "refactoring error handling" or similar text :-)